client/src/main/java/org/apache/cassandra/sidecar/client/exception/UnexpectedStatusCodeException.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.client.exception;
+
+/**
+ * Represents an exception raised when an unexpected status code is encountered
+ */
+public class UnexpectedStatusCodeException extends RuntimeException
+{
+    /**
+     * Constructs a new {@link UnexpectedStatusCodeException} with the given {@code statusCode} and {@code body}
+     *
+     * @param statusCode the response status code
+     * @param body       the body of the response
+     */
+    public UnexpectedStatusCodeException(int statusCode, String body)
```

Comment:
Is it safe to assume the body is always `String`?

Reply:
good point. I think this can be a binary file that we stream

Reply:
this constructor is called from 
```
    UnexpectedStatusCodeException unexpectedStatusCode(HttpResponse response)
    {
        return new UnexpectedStatusCodeException(response.statusCode(), response.contentAsString());
    }
```
and `response.contentAsString()` is a best effort attempt to convert the raw response to a UTF-8 String,

Reply:
will leave for now as it's an unexpected case, what do you think?